### ISSUE TITLE
Assertion failed in sd-dhcp-lease.c on OpenSUSE

### ISSUE THREAD

**Issue by mrc0mmand** [systemd/systemd#21671]
**systemd version the issue has been seen with**
 > latest main

**Used distribution**
 > OpenSUSE

In the recent PRs I noticed that the OpenSUSE mkosi CI sometimes fails due to a timeout. I tried to track down the issue and noticed that systemd-networkd keep dying with SIGABRT:

```
2021-12-07T20:03:20.0283681Z [   57.767682] systemd-networkd[393]: enp0s1: DHCPv4 client: OFFER
2021-12-07T20:03:20.0284296Z 
2021-12-07T20:03:20.0286514Z varlink-46: New incoming message: {"method":"io.systemd.UserDatabase.GetMemberships","parameters":{"groupName":"wheel","service":"io.systemd.DynamicUser"},"more":true}
2021-12-07T20:03:20.0291646Z 
2021-12-07T20:03:20.0335523Z varlink-46: Changing state idle-server → processing-method-more
2021-12-07T20:03:20.0336497Z 
2021-12-07T20:03:20.0446714Z varlink-46: Sending message: {"error":"io.systemd.UserDatabase.NoRecordFound","parameters":{}}
2021-12-07T20:03:20.0448033Z 
2021-12-07T20:03:20.0449357Z [   57.780388] systemd-networkd[393]: enp0s1: link_check_ready(): IPv6LL is not configured yet.
2021-12-07T20:03:20.0450108Z 
2021-12-07T20:03:20.0451138Z [   57.790653] systemd-networkd[393]: enp0s1: DHCPv4 client: REQUEST (requesting)
2021-12-07T20:03:20.0451835Z 
2021-12-07T20:03:20.0452981Z varlink-46: Changing state processing-method-more → processed-method
2021-12-07T20:03:20.0453794Z 
2021-12-07T20:03:20.0487448Z varlink-46: Changing state processed-method → idle-server
2021-12-07T20:03:20.0488237Z 
2021-12-07T20:03:20.0517605Z [   57.793243] systemd-networkd[393]: Failed to open timezone file '/usr/share/zoneinfo/Etc/UTC': No such file or directory
2021-12-07T20:03:20.0518465Z 
2021-12-07T20:03:20.0520052Z [   57.796017] systemd-networkd[393]: Assertion '(_error) != 0' failed at src/libsystemd-network/sd-dhcp-lease.c:778, function dhcp_lease_parse_options(). Aborting.
```

Full log: 
[opensuse-mkosi.log](https://github.com/systemd/systemd/files/7671473/opensuse-mkosi.log)

I'm not sure if it's the direct cause of the timeouts, but it looks like something to fix. /cc @yuwata, @Werkov

**Comment by yuwata** [comment]
Fix is waiting in #21674.

**Comment by yuwata** [comment]
Note that issue is caused by 0339cd7707ac40c29d51ee6933d06bc87ed140a4.
